astropy/coordinates/angles.py

```diff
@@ -632,7 +632,11 @@ def _wrap_internal(self):
         wrap_angle_floor = wrap_angle - a360
         self_angle = self.value
         # Do the wrapping, but only if any angles need to be wrapped
-        if np.any(self_angle < wrap_angle_floor) or np.any(self_angle >= wrap_angle):
+        # Make sure NaNs don't emit a warning
+        with np.errstate(invalid='ignore'):
```

Comment:
An alternative might be to avoid the comparison (which is really rather ugly!). How about the following (slightly faster too! and a bit clearer, I think):
```
         wrap_angle_floor = wrap_angle - a360
         self_angle = self.value
         # Do the wrapping, but only if any angles need to be wrapped
-        if np.any(self_angle < wrap_angle_floor) or np.any(self_angle >= wrap_angle):
-            wrapped = np.mod(self_angle - wrap_angle, a360) + wrap_angle_floor
-            value = u.Quantity(wrapped, self.unit)
-            super().__setitem__((), value)
+        wraps = (self_angle - wrap_angle_floor) // a360
+        if np.any(wraps != 0):
+            self -= (wraps * a360) << self.unit
```

Reply:
p.s. I'm actually not all that sure that it is really necessary to convert quantities to arrays...  But out of scope here!